pinot-controller/src/main/java/com/linkedin/pinot/controller/ControllerStarter.java

```diff
@@ -148,6 +153,40 @@ public Long call() throws Exception {
               }
             });
 
+    controllerMetrics.addCallbackGauge("dataDir.exists", new Callable<Long>() {
+      @Override
+      public Long call() throws Exception {
+        return new File(config.getDataDir()).exists() ? 1L : 0L;
+      }
+    });
+
+    controllerMetrics.addCallbackGauge("dataDir.fileOpLatencyMs", new Callable<Long>() {
+      @Override
+      public Long call() throws Exception {
+        File dataDir = new File(config.getDataDir());
+
+        if (dataDir.exists()) {
+          try {
+            long startTime = System.currentTimeMillis();
+            final File testFile = new File(dataDir, config.getControllerHost());
+            FileOutputStream outputStream = new FileOutputStream(testFile, false);
+            outputStream.write(Longs.toByteArray(System.currentTimeMillis()));
+            outputStream.flush();
+            outputStream.close();
+            FileUtils.deleteQuietly(testFile);
+            long endTime = System.currentTimeMillis();
+
+            return endTime - startTime;
+          } catch (IOException e) {
+            LOGGER.warn("Caught exception while checking the data directory operation latency", e);
+            return null;
```

Comment:
i think returning a high value of this metric may be better in this case (and in the case when datadir does not exist)